csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs

```diff
@@ -764,9 +764,7 @@ private static OrtValue ExtractOrtValueForOutput(NamedOnnxValue output, NodeMeta
                 }
                 if (ortValue != null)
                 {
-                    if (ortValue.IsOwned)
-                        cleanupList.Add(ortValue);
-
+                    cleanupList.Add(ortValue);
```

Comment:
It is not always owned. However, in either case, Dispose() would do the right thing. I made this change because Dispose() would also suppress finalization.